### PR TITLE
PMM-13543 Support socket path.

### PR DESCRIPTION
**Pull request by BupycHuk**
[PMM-13477](https://jira.percona.com/browse/PMM-13477) Support of PSMDB 8.0.

[PMM-13543](https://jira.percona.com/browse/PMM-13543) Support socket path.

- [ ] Links to related pull requests (optional).
- https://github.com/percona/pmm/pull/3308

---

Below we provide a basic checklist of things that would make it a good PR:
- Make sure to sign the CLA (Contributor License Agreement).
- Make sure all tests pass.
- Keep current with the target branch and fix conflicts if necessary.
- Update jira ticket description if necessary.
- Attach screenshots and/or console output to the jira ticket to confirm new behavior, if applicable.
- Leave notes to the reviewers if you need to focus their attention on something specific.

Once all checks pass and the code is ready for review, please add `pmm-review-exporters` team as the reviewer. That would assign people from the review team automatically. Report any issues on our [Forum](https://forums.percona.com).

[PMM-13477]: https://perconadev.atlassian.net/browse/PMM-13477?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[PMM-13543]: https://perconadev.atlassian.net/browse/PMM-13543?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: exporter/diagnostic_data_collector.go [hunk 1]
```diff
@@ -117,7 +117,7 @@ func (d *diagnosticDataCollector) collect(ch chan<- prometheus.Metric) {
 		debugResult(logger, m)
 
 		// MongoDB 8.0 splits the diagnostic data into multiple blocks, so we need to merge them
-		if d.buildInfo.VersionArray[0] >= 8 { //nolint:gomnd
+		if _, ok := m["common"]; ok { //nolint:gomnd,mng
```
Comment (ademidoff) [inline exporter/diagnostic_data_collector.go:120]:
```suggestion
		if _, ok := m["common"]; ok { //nolint:gomnd,mnd
```

File: main.go [hunk 2]
```diff
@@ -268,18 +268,36 @@ func parseURIList(uriList []string, logger *logrus.Logger, splitCluster bool) []
 	return URIs
 }
 
+// buildURIManually builds the URI manually by checking if the user and password are supplied
+func buildURIManually(uri string, user string, password string) string {
+	uriArray := strings.SplitN(uri, "://", 2)
```
Comment (ademidoff) [inline main.go:273]:
```suggestion
	uriArray := strings.SplitN(uri, "://", 2) //nolint:mnd
```

**Review by ademidoff — APPROVED** [review]

**Review by JiriCtvrtka — APPROVED** [review]

File: exporter/diagnostic_data_collector.go (hunk 1 above)
Comment (ademidoff) [reply exporter/diagnostic_data_collector.go]: Even better :)

Comment (BupycHuk) [reply exporter/diagnostic_data_collector.go]: thanks, it was stupid from my side. I realized, that I don't use magic number there anymore, so I can drop that nolint

File: main.go [hunk 3]
```diff
@@ -268,18 +268,36 @@ func parseURIList(uriList []string, logger *logrus.Logger, splitCluster bool) []
...
 
+// buildURIManually builds the URI manually by checking if the user and password are supplied
+func buildURIManually(uri string, user string, password string) string {
+	uriArray := strings.SplitN(uri, "://", 2) //nolint:mnd
+	prefix := uriArray[0] + "://"
+	uri = uriArray[1]
+
+	// IF user@pass not contained in uri AND custom user and pass supplied in arguments
+	// DO concat a new uri with user and pass arguments value
+	if !strings.Contains(uri, "@") && user != "" && password != "" {
```
Comment (ademidoff) [inline main.go:279]: I'm wondering if we shouldn't support a case where the user is non-empty and the password is empty.

Comment (BupycHuk) [reply main.go:279]: good point for improvement, but in scope of this fix I think we can leave as it used to be. I just reverted old logic for cases where url.Parse doesn't work. Additionally for that users can set user in initial DSN. let's create github issue with good-first-task label for those who would like to contribute

**Pull request merged**